locust/clients.py

```diff
@@ -130,10 +130,16 @@ def request(self, method, url, name=None, catch_response=False, context={}, **kw
         # prepend url with hostname unless it's already an absolute URL
         url = self._build_url(url)
 
+        # use stream=True to avoid closing socket and downloading content at the beginning
+        # we'll download content if actually requested a bit later
+        #
+        # a bit of a hack to actually estimate fetching time
+        stream = kwargs.pop("stream", False)
```

Comment:
@cyberw I really hate this one, but it seems standard `requests.Session` fetches the content eagerly when `stream=False` is used. So the only way to actually estimate `fetching` is to use `stream=True` during first request. Then consume `content` normally later.

Reply:
Does setting stream=True affect the headers sent, as this implies?https://stackoverflow.com/questions/60343944/how-does-requests-stream-true-option-streams-data-one-block-at-a-time
If it does then I dont think this change is ok...